libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3310,23 +3350,37 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 					toCategory == Type::Category::Integer ?
 					dynamic_cast<IntegerType const&>(_to) :
 					addressType;
-
-				// Clean according to the "to" type, except if this is
-				// a widening conversion.
-				IntegerType const* cleanupType = &to;
-				if (fromCategory != Type::Category::RationalNumber)
+				if (fromCategory == Type::Category::FixedPoint)
 				{
-					IntegerType const& from =
-						fromCategory == Type::Category::Integer ?
-						dynamic_cast<IntegerType const&>(_from) :
-						addressType;
-					if (to.numBits() > from.numBits())
-						cleanupType = &from;
+					auto const& fixedFrom = dynamic_cast<FixedPointType const&>(_from);
+					solAssert(fixedFrom.isSigned() == to.isSigned(), "" );
+					body =
+						Whiskers("converted := <cleanInt>(<div>(<cleanFixed>(value), <digits>))")
+						("cleanFixed", cleanupFunction(_from))
+						("div", to.isSigned() ? "sdiv" : "div")
+						("digits", "1" + string{fixedFrom.fractionalDigits(), '0'})
```

Comment:
`clang` seems to not like this. `("digits", "1" + string(fixedFrom.fractionalDigits(), '0'))` solves the compilation issue.

```
/Users/alex/git/solidity/libsolidity/codegen/YulUtilFunctions.cpp:3361:31: error: non-constant-expression cannot be narrowed from type 'unsigned int' to 'char' in initializer list [-Wc++11-narrowing]
                                                ("digits", "1" + string{fixedFrom.fractionalDigits(), '0'})
                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/alex/git/solidity/libsolidity/codegen/YulUtilFunctions.cpp:3361:31: note: insert an explicit cast to silence this issue
                                                ("digits", "1" + string{fixedFrom.fractionalDigits(), '0'})
                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                                        static_cast<char>(          )
/Users/alex/git/solidity/libsolidity/codegen/YulUtilFunctions.cpp:3361:41: error: implicit conversion loses integer precision: 'unsigned int' to 'const char' [-Werror,-Wimplicit-int-conversion]
                                                ("digits", "1" + string{fixedFrom.fractionalDigits(), '0'})
                                                                       ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
2 errors generated.
```

Reply:
Interesting that `b_osx` didn't had any problems. I guess we use there a slightly older clang version.

```
➜  clang --version
Apple clang version 12.0.5 (clang-1205.0.22.11)
Target: arm64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```